src/groups/mqb/mqba/mqba_application.cpp

```diff
@@ -560,17 +559,19 @@ bool Application::initiateShutdown()
                 ? negoMsg.clientIdentity()
                 : negoMsg.brokerResponse().brokerIdentity();
 
-        bool isBroker = false;
+        bool isClusterNode = false;
+        bool isProxy       = false;
         if (mqbnet::ClusterUtil::isClientOrProxy(negoMsg)) {
             clients.push_back(sessionSp);
+
             if (!negoMsg.clientIdentity().clusterName().empty()) {
-                isBroker = true;
+                isProxy = true;
             }
         }
         else {
-            isBroker = true;
+            isClusterNode = true;
```

Comment:
Can revert to use `isBroker` and push nodes in-place. It will simplify the code by removing late `if` 
```suggestion
            isBroker = true;
            nodes.push_back(sessionSp);
```